source/base/hdf5.cc

```diff
@@ -410,18 +408,17 @@ namespace HDF5
     int    int_value;
     hid_t  attr;
     herr_t ret;
+    (void)ret;
 
     attr = H5Aopen(*hdf5_reference, attr_name.data(), H5P_DEFAULT);
     Assert(attr >= 0, ExcMessage("Error at H5Aopen"));
     ret = H5Aread(attr, H5T_NATIVE_INT, &int_value);
     Assert(ret >= 0, ExcMessage("Error at H5Aread"));
     ret = H5Aclose(attr);
     Assert(ret >= 0, ExcMessage("Error at H5Aclose"));
-    // The int can be casted to a bool
-    bool bool_value = (bool)int_value;
 
-    (void)ret;
-    return bool_value;
+    // The int can be casted to a bool
+    return (int_value != 0);
```

Comment:
you don't actually need the comment any more, as there is no more cast happening here :-)